proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -562,9 +563,17 @@ public void disconnect(String reason)
     @Override
     public void disconnect(final BaseComponent... reason)
     {
-        if ( thisState != State.STATUS && thisState != State.PING )
+        if ( thisState != State.STATUS && thisState != State.PING && thisState != State.HANDSHAKE )
         {
-            ch.delayedClose( new Kick( ComponentSerializer.toString( reason ) ) );
```

Comment:
I suggest readding the comment https://github.com/SpigotMC/BungeeCord/pull/2722/files#diff-d81a4f22970536a8f3cfe44bfd87df6eL106-L108

Reply:
I suggest reading the code after that line